src/files/infrastructure/FileJSDataverseRepository.ts

```diff
@@ -77,6 +82,17 @@ export class FileJSDataverseRepository implements FileRepository {
       })
   }
 
+  getFilesTotalDownloadSizeByDatasetPersistentId(
+    datasetPersistentId: string,
+    datasetVersion: DatasetVersion
+  ): Promise<number> {
+    return getDatasetFilesTotalDownloadSize
+      .execute(datasetPersistentId, datasetVersion.toString(), FileDownloadSizeMode.ARCHIVAL)
+      .catch((error: WriteError) => {
```

Comment:
I added 2 commits 

1. I found that the error when running the dev-env was because the package-lock.json wasn't being copied to the docker image so there were some conflicts between the dependencies because npm was using the last version of the libraries. I added a commit to fix this
2. You're right about the error types, I fixed it everywhere and I'll put more attention next time when adding those types, thanks!